CHANGELOG.rst

```diff
@@ -5,7 +5,100 @@ Changelog
+    class ProductImagePage(WebPage[Image]):
+        @field
+        def url(self) -> str:
+            return self.css("#product img ::attr(href)").get("")
+
+    @attrs.define
+    class Product:
+        name: str
+        image: Image
+
+    @handle_urls("example.com")
+    @attrs.define
+    class ProductPage(WebPage[Product]):
+        # ✨ NEW: Notice that the page object can ask for items as dependencies.
+        # An instance of ``Image`` is injected behind the scenes by calling the
+        # ``.to_item()`` method of ``ProductImagePage``.
+        image_item: Image
+
+        @field
+        def name(self) -> str:
+            return self.css("h1.name ::text").get("")
+
+        @field
+        def image(self) -> Image:
+            return self.image_item
+
+    class MySpider(scrapy.Spider):
+        name = "myspider"
+
+        def start_requests(self):
+            yield scrapy.Request(
+                "https://example.com/products/some-product", self.parse
+            )
+
+        # ✨ NEW: Notice that we're directly using the item here and not the
+        # page object.
+        def parse(self, response: DummyResponse, item: Product):
+            return item
+
+In line with this, the following changes were made:
+
+    * Added a new ``scrapy_poet.page_input_providers.ItemProvider`` which makes
+      the usage above possible.
+    * Multiple changes to the ``scrapy_poet.PageObjectInputProvider`` base class
+      which are backward incompatible:
+
+        * It now accepts instance of ``scrapy_poet.injection.Injector`` in its
+          constructor instead of ``scrapy.crawler.Crawler``. Although you can
+          still access the ``scrapy.crawler.Crawler`` via ``Injector.crawler``
+          attribute.
+        * The ``is_provided()`` is now an instance method instead of a class
+          method.
+
+    * An item type is now supported by ``scrapy_poet.callback_for`` alongside
+      the usual page objects. This means that it won't raise a ``TypeError``
+      anymore when not passing a subclass of ``web_poet.ItemPage``.
+    * ``scrapy_poet.overrides.OverridesRegistry`` has been overhauled:
+
+        * It is now subclassed from ``web_poet.RulesRegistry`` which allows
+          outright access to its registry methods.
+        * It now allows retrieval of rules based on the returned item class.
+        * ``OverridesRegistry`` (alongside ``SCRAPY_POET_OVERRIDES``) won't
+          accept tuples as rules anymore. Only ``web_poet.ApplyRule``
+          instances are allowed.
+
+            * This also results from these type aliases to be removed:
+              ``scrapy_poet.overrides.RuleAsTuple`` and
+              ``scrapy_poet.overrides.RuleFromUser``
+            * These changes are backward incompabible.
+
+    * New exception: ``scrapy_poet.injector_error.ProviderDependencyDeadlockError``.
+      This is raised when it's not possible to create the dependencies due to
+      a deadlock in their sub-dependencies.
```

Comment:
I think we might want to be explicit in mentioning circular dependencies as the cause of the deadlock. For example:

```suggestion
      This is raised when it's not possible to create the dependencies due to
      a deadlock in their sub-dependencies, e.g. due to a circular dependency
      between page objects.
```